queries/liquid/highlights.scm

```diff
@@ -0,0 +1,123 @@
+([
+  "{{"
+  "}}"
+  "{{-"
+  "-}}"
+  "{%"
+  "%}"
+  "{%-"
+  "-%}"
+] @tag.delimiter
+  (#set! priority 110))
```

Comment:
Is this still intended for all highlights to be 110 priority? 🤔

Reply:
Also, it doesn't seem like these ranges are ever injected, so it should mean priorities here are unnecessary

Reply:
Because of this language is embedded and injections are done with `injection.combined` this is (afaik) the only way I can see to achieve the correct highlighting, as I eluded to @clason above.

To clarify,  highlight groups the sometimes conflict with injected "parent" language. For example, JavaScript highlights will be prioritized over the Liquid highlights when Liquid is embedded in a `<script>` tag. 

What you pointed out makes a perfect example of what I'm saying here.
With priority 110 (how its intended to be highlighted):
<img width="491" alt="Screenshot 2024-03-27 at 10 36 26 PM" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/38442220/524e24fc-27d8-4ee4-a261-fe79efd28515">

Without any priority:
<img width="477" alt="Screenshot 2024-03-27 at 10 52 12 PM" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/38442220/872502e6-2145-49c7-b59d-c8c57b929670">

Reply:
but this should mean the js tree is broken or not 🤔

Reply:
Yes, unfortunately this does mean the js tree (or any injected tree for that matter) is broken during these portions. Frankly, I don't really see a way around this using `injection.combined` without writing the parser to explicitly include the language constructs for html/js/css without injection (open to suggestions ofc -- I am new to ts and may be missing something..).

That being said, I haven't encountered any issues with the injected languages rendered incorrectly in terms of highlights, and since `injection.combined` is implemented upstream I honestly assumed this tree breaking behavior was to be expect when using injected languages this way.

Reply:
I did a bit more digging into this issue. I think that because the html parser injects js itself, js trys to parse the entire range within the script tag, rather than the combined `template_content` nodes..

you can see here that the html content is correctly captured and parsed..
<img width="658" alt="Screenshot 2024-03-28 at 10 03 03 AM" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/38442220/22667e7b-fff3-4c47-8919-7b7b85849120">

but here, the js/script content is correctly captured, but not parsed as such -- all the content in the script tag is attempted to be parsed rather than just the combined `template_content` nodes like in its html counterpart.
<img width="821" alt="Screenshot 2024-03-28 at 10 02 32 AM" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/38442220/b5bda262-b1d9-4a8d-a9fd-0aa76d52f772">
<img width="850" alt="Screenshot 2024-03-28 at 10 27 47 AM" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/38442220/da852f0a-9316-4105-b149-213b439d5003">

It's a bit messy. Still, the only way I see around this is to more explicitly handle how the html is parsed in the template, and avoid leaning on just injecting the html parser over all template content.

Reply:
Yeah definitely.  It certainly does feel a bit like a "putting a square peg in a round hole" type of situation lol.

Anyways, I'd be willing to try some alternatives out I suppose, but I do feel using injections in this way is the most logical and (maybe?) idiomatic approach given the tools. Other template languages here do the same and suffer from the same issue, it just may not have been quite as apparent or investigated.

For example, the Vento language (added recently to this repo) also cannot correctly parse when dropping into a script tag either: 
<img width="841" alt="Screenshot 2024-03-28 at 1 45 26 PM" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/38442220/5b215c1d-f83e-45fa-b889-f4bf8d9f9e35">

Same with htmldjango (though experimental).

I guess in concussion to me it makes sense to make a concession with the higher priorities and at least have it appear to be correct.  But, I do understand if breaking the tree of an injected language is not acceptable.  Like you said I do think it's ultimately a just hard problem bc upstream implementation.

Reply:
Yes, this applies to all template languages, and my primary concern is to have a _consistent_ sensible strategy (given the constraints, which we can't do anything about).

On the other hand, I do not care so much about correctness; our goal here is syntax highlighting, not a syntax tree for code validation or transformation.

So priorities are not a deal-breaker (especially if this _is_ the consistent strategy), but should be used surgically, so only if absolutely necessary (and the check is per pattern).

Reply:
I removed what I see to be the unnecessary highlights after doing some testing, and left the remaining queries at 110 priority.